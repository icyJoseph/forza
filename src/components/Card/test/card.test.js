import React from "react";
import { shallow } from "enzyme";
import { CardWrap, Wrap, Card } from "../";

describe("CardWrap", () => {
  const wrapperWithProps = shallow(<CardWrap height={100} />);
  const wrapper = shallow(<CardWrap />);

  it("renders", () => {
    expect(wrapper).toBeTruthy();
  });
  it("gets auto height", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("gets defined height", () => {
    expect(wrapperWithProps.html()).toMatchSnapshot();
  });
});

describe("Wrap", () => {
  const props = {
    title: 0,
    index: 0,
    highlight: -1
  };
  const otherProps = {
    title: 1,
    index: 0,
    highlight: 1
  };
  const wrapper = shallow(<Wrap {...props} />);
  const wrapper2 = shallow(<Wrap {...otherProps} />);
  it("gets fontSize 14pt and Color dodgerblue", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("gets fontSize 12pt and Color black", () => {
    expect(wrapper2).toMatchSnapshot();
  });
});

describe("Card", () => {
  const children = [<div>hi</div>, <div>hi</div>];
  const props = {
    handler: jest.fn(),
    id: "aa",
    current: "aa"
  };
  const wrapper = shallow(<Card {...props} children={children} />);
  it("renders two children", () => {
    expect(wrapper.children()).toHaveLength(2);
  });
  it("uses the handler", () => {
    wrapper.simulate("click");
    expect(props.handler).toHaveBeenCalled();
  });
  it("uses the correct elevation for id === current", () => {
    expect(wrapper.props()).toHaveProperty("elevation", 24);
  });
  it("uses the correct elevation for id!== current", () => {
    const otherProps = {
      ...props,
      current: "aaaa"
    };
    const wrapper2 = shallow(<Card {...otherProps} children={children} />);
    expect(wrapper2.props()).toHaveProperty("elevation", 3);
  });
});
