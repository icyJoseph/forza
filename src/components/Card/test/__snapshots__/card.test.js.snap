// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardWrap gets auto height 1`] = `"<div class=\\"MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa ffcFVI\\" style=\\"height:auto\\"></div>"`;

exports[`CardWrap gets defined height 1`] = `"<div class=\\"MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 sc-bdVaJa ffcFVI\\" style=\\"height:100px\\" height=\\"100\\"></div>"`;

exports[`Wrap gets fontSize 12pt and Color black 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div
    highlight={1}
    index={0}
    title={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-htpNat ffBkYG",
      "style": Object {
        "color": "dodgerblue",
        "fontSize": "12pt",
      },
      "title": 1,
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-htpNat ffBkYG",
        "style": Object {
          "color": "dodgerblue",
          "fontSize": "12pt",
        },
        "title": 1,
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Wrap gets fontSize 14pt and Color dodgerblue 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div
    highlight={-1}
    index={0}
    title={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-htpNat ffBkYG",
      "style": Object {
        "color": "black",
        "fontSize": "14pt",
      },
      "title": 0,
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-htpNat ffBkYG",
        "style": Object {
          "color": "black",
          "fontSize": "14pt",
        },
        "title": 0,
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
